wazo_calld/plugin_helpers/confd.py

```diff
@@ -119,15 +121,25 @@ def exists(self):
 
     @classmethod
     def from_id(cls, conference_id, confd_client):
+        conference = cls._get_by_id_or_none(conference_id, confd_client)
+        if not conference:
+            raise NoSuchConferenceID(conference_id)
+
+        return cls(conference['tenant_uuid'], conference['id'], confd_client)
+
+    @classmethod
+    def reset_cache(cls, _):
+        cls._get_by_id_or_none.cache_clear()
+
+    @classmethod
+    @lru_cache(maxsize=128)
+    def _get_by_id_or_none(cls, conference_id, confd_client):
```

Comment:
lru_cache will not cache exceptions. 404 is caught here and returned as None, all other exceptions are raised to avoid caching.